core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -741,9 +741,10 @@ abstract class RDD[T: ClassTag](
   def mapWith[A, U: ClassTag]
       (constructA: Int => A, preservesPartitioning: Boolean = false)
       (f: (T, A) => U): RDD[U] = withScope {
+    val cleanF = sc.clean(f)
     mapPartitionsWithIndex((index, iter) => {
       val a = constructA(index)
-      iter.map(t => f(t, a))
+      iter.map(t => cleanF(t, a))
```

Comment:
my understanding is that mappartitionsWithIndex cleans, and thus anything in that is cleaned. maybe it's an incorrect assumption. cc @andrewor14?